Alignment/OfflineValidation/bin/GCPtrends.cc

```diff
@@ -0,0 +1,275 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include "TFile.h"
+#include "TChain.h"
+#include "TSelector.h"
+#include "TTreeReader.h"
+#include "TTreeReaderValue.h"
+#include "TTreeReaderArray.h"
+#include "TGraphErrors.h"
+#include "TH1D.h"
+#include "TObject.h"
+#include "TMath.h"
+#include "TString.h"
+
+#include "Alignment/OfflineValidation/interface/Trend.h"
+
+int main(int argc, char *argv[]) {
+  if (argc < 2) {
+    std::cout << "Usage of the program : GCPtrends [gcp_trend_config.json]" << std::endl;
+    std::cout << "gcp_trend_config.json : file with configuration in json format" << std::endl;
+    exit(1);
+  }
+
+  // parsing input file
+  std::string lumi_file(argv[1]);
+  boost::property_tree::ptree config_root;
+  boost::property_tree::read_json(lumi_file, config_root);
+
+  // configuration
+  TString outputdir = config_root.get<std::string>("outputdir", ".");
+  outputdir = TString(outputdir) + TString(outputdir.EndsWith("/") ? "" : "/");
+
+  bool usebadmodules = config_root.get<bool>("usebadmodules", true);
+
+  bool splitpixel = config_root.get<bool>("splitpixel", false);
+
+  Int_t trackerphase = config_root.get<int>("trackerphase");
+
+  TString lumitype = config_root.get<std::string>("trends.lumitype");
+
+  Int_t firstrun = config_root.get<int>("trends.firstrun");
+
+  Int_t lastrun = config_root.get<int>("trends.lastrun");
+
+  std::string lumibyrunfile = config_root.get<std::string>("lumibyrunfile", "");
+  std::ifstream lumifile(lumibyrunfile);
+  assert(lumifile.good());
+  const Run2Lumi lumi_per_run(lumibyrunfile, firstrun, lastrun, 1000);
+
+  std::map<TString, Int_t> comparison_map;
+  for (boost::property_tree::ptree::value_type &comparison : config_root.get_child("comparisons")) {
+    TString path = comparison.first;
+    Int_t run_label = comparison.second.get_value<int>();
+
+    comparison_map[path] = run_label;
+  }
+
+  // create output file
+  TFile *file_out = TFile::Open(outputdir + "GCPTrends.root", "RECREATE");
+  file_out->cd();
+
+  gErrorIgnoreLevel = kError;
+
+  // sub-detector mapping
+  std::map<Int_t, TString> Sublevel_Subdetector_Map = {
+      {1, "PXB"}, {2, "PXF"}, {3, "TIB"}, {4, "TID"}, {5, "TOB"}, {6, "TEC"}};
+
+  // wheels/layers per tracker phase: <pahse,<sublevel,number of layers/wheels>>
+  const std::map<Int_t, std::map<Int_t, Int_t>> Phase_Subdetector_Layers_Map = {
+      {0, {{1, 3}, {2, 2}}}, {1, {{1, 4}, {2, 3}}}, {2, {{1, 4}, {2, 12}}}};
```

Comment:
so in the end you dont plan to have this through the topology, right?

Reply:
this is essentially a plotting macro, and it doesn't have access to `cmssw` EventSetup. I guess that if you really insist, one could build the topology using the `StandaloneTrackerTopology` class using different hard-coded versions of the `TrackerParameters.xml` depending on the phase. I am not against it, but looks perhaps an unnecessary complication.

Reply:
No, I think it would be over-killing it for a simple 2D mapping. Moreover, this macros is currently in an almost-cmssw-free standalone version, only requiring ROOT & boost, it'd be preferable to leave it simple as that considering it works on standard ROOT TTrees

Reply:
ok, I'll sign, thanks!